libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1066,9 +1085,35 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 		solAssert(false, "Index access only allowed for mappings or arrays.");
 }
 
-void IRGeneratorForStatements::endVisit(IndexRangeAccess const&)
-{
-	solUnimplementedAssert(false, "Index range accesses not yet implemented.");
+void IRGeneratorForStatements::endVisit(IndexRangeAccess const& _indexRangeAccess) {
+	Type const& baseType = *_indexRangeAccess.baseExpression().annotation().type;
+	solUnimplementedAssert(
+		baseType.category() == Type::Category::Array || baseType.category() == Type::Category::ArraySlice,
+		"Index range accesses is available only on arrays and array slices."
+	);
+	solAssert(_indexRangeAccess.startExpression(), "Start index expression expected.");
+	solAssert(_indexRangeAccess.endExpression(), "End index expression expected.");
```

Comment:
Both can be empty expressions, so needs to fix this

Also, add more tests to cover these